h2o-py/tests/pyunit_utils/utilsPY.py

```diff
@@ -3431,4 +3432,192 @@ def random_dataset_numeric_only(nrow, ncol, integerR=100, misFrac=0.01):
 
 def getMojoName(modelID):
     regex = re.compile("[+\\-* !@#$%^&()={}\\[\\]|;:'\"<>,.?/]")
-    return regex.sub("_", modelID)
\ No newline at end of file
+    return regex.sub("_", modelID)
+
+
+def genDMatrix(h2oFrame, yresp, enumCols=[]):
+    """
+    This method will convert a H2OFrame containing all numerical dataset to a DMatrix that is used by native XGBoost.
+    We added an extra level of missing(NA) column.  Note that we do not deal with NA/missing values here
+
+    :param h2oFrame:
+    :param xlist:
+    :param yresp:
+    :return:
+    """
+    import xgboost as xgb
+
+    pandaFtrain = h2oFrame.as_data_frame(use_pandas=True, header=True)
+    nrows = h2oFrame.nrow
+
+    if len(enumCols) > 0:   # start with first enum column
+        pandaTrainPart = generatePandaEnumCols(pandaFtrain, enumCols[0], nrows)
+        pandaFtrain.drop([enumCols[0]], axis=1, inplace=True)
+
+        for colInd in range(1, len(enumCols)):
+            cname=enumCols[colInd]
+            ctemp = generatePandaEnumCols(pandaFtrain, cname,  nrows)
+            pandaTrainPart=pd.concat([pandaTrainPart, ctemp], axis=1)
+            pandaFtrain.drop([cname], axis=1, inplace=True)
+
+        pandaFtrain = pd.concat([pandaTrainPart, pandaFtrain], axis=1)
+
+    c0= h2oFrame[yresp].asnumeric().as_data_frame(use_pandas=True, header=True)
+    pandaFtrain.drop([yresp], axis=1, inplace=True)
+    pandaF = pd.concat([c0, pandaFtrain], axis=1)
+    pandaF.rename(columns={c0.columns[0]:yresp}, inplace=True)
+    newX = list(pandaFtrain.columns.values)
+    data = pandaF.as_matrix(newX)
+    label = pandaF.as_matrix([yresp])
+
+    return xgb.DMatrix(data=data, label=label)
+
+def generatePandaEnumCols(pandaFtrain, cname, nrows):
+    cmissingNames=[cname+".missing(NA)"]
+    tempnp = np.zeros((nrows,1), dtype=np.int)
+    # check for nan and assign it correct value
+    colVals = pandaFtrain[cname]
+    for ind in range(nrows):
+        try:
+            float(colVals[ind])
+            if math.isnan(colVals[ind]):
+                tempnp[ind]=1
+        except ValueError:
+            pass
+    zeroFrame = pd.DataFrame(tempnp)
+    zeroFrame.columns=cmissingNames
+    temp = pd.get_dummies(pandaFtrain[cname], prefix=cname, drop_first=False)
+    ctemp=pd.concat([temp,zeroFrame], axis=1)   # transformed H2O enum column to panda enum columns plus NA column
+    return ctemp
+
+
+def genDMatrix2(h2oFrame, yresp, enumCols=[]):
```

Comment:
The naming should be better in this case. The documentation says it all, but let's rename it to `genDMatrixNumerical `.

Reply:
Actually, the documentation is off.  I need to fix it.  Changed the name to transformH2OFrameToDMatrix